cmd/sample-generic-controlplane/server/config.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	apiextensionsapiserver "k8s.io/apiextensions-apiserver/pkg/apiserver"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/util/webhook"
+	aggregatorapiserver "k8s.io/kube-aggregator/pkg/apiserver"
+	aggregatorscheme "k8s.io/kube-aggregator/pkg/apiserver/scheme"
+	"k8s.io/kubernetes/pkg/controlplane"
+
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
```

Comment:
do we plan to extract the rest packages of the "generic" core resources (like secrets, crds, etc.) and put them somewhere to cut the k8s.io/kubernetes dependency, or is that not a goal?

if so, we'd probably (?) want to make a scheme specifically for these "core/v1 types that are generic"

Reply:
One of the big questions (with many opinions, we will see) to discuss.